File: leet/58_lengthOfLastWord.ts
// https://leetcode.com/problems/length-of-last-word/description/

function lengthOfLastWord(s: string): number {
    // Split on whitespace and filter out whitespace substrings
    const words = s.split(" ").filter(i => i !== "") 

    return words[words.length - 1].length
};


function lengthOfLastWord_LF(s: string): number {
    const words = s.trimEnd().split(" ") 
    const lastWord = words.pop()
    return lastWord ? lastWord.length : 0
  };

console.log(lengthOfLastWord("hello world"))
console.log(lengthOfLastWord("hello    world     "))
console.log(lengthOfLastWord_LF("hello world"))
console.log(lengthOfLastWord_LF("hello    world     "))

File: 58_lengthOfLastWord.c
// https://leetcode.com/problems/length-of-last-word/description/

int lengthOfLastWord(char* s) {
    int found_space = 0;
    int l = 0;
    for (int i=0;i<strlen(s); i++){
        if (s[i]==' '){
            found_space = 1;
            continue;
        } else if (found_space) {
            found_space = 0;
            l = 0;
        }
        
        l++;        
    }
    return l;
}